accepted/2021/x64-emulation-on-arm64/x64-emulation.md

```diff
@@ -0,0 +1,243 @@
+- [RFC: dotnet [Arm64 | x64] coexistence](https://github.com/dotnet/sdk/issues/16896)
+
+## Install location
+
+Goal: .NET is installed to a disk location(s) that makes sense across time and space.
+
+We have to install the .NET x64 product *somewhere* on Arm64 machines. We can install it in a `dotnet-x64` directory (or similar variant), as a peer to `dotnet` or as child of the `dotnet` directory within a new `x64` directory. We decided on the latter. The former sticks out like a sore thumb. The layout within the `x64` directory will be the exact same as a native architecture x64 installation.
+
+The expected install location of x64 .NET on Arm64:
+
+- macOS: `/usr/local/share/dotnet/x64`
+- Windows: `C:\Program Files\dotnet\x64`
+
+There are a few implications of this decision.
+
+- We'll only add the parent `dotnet` directory to the `PATH`, which means that only the native architecture installation will be usable when you type `dotnet`.
+- It will be possible for a machine to have .NET x64 installed by itself, with no Arm64 .NET. That means that the `dotnet` directory would be empty, with the exception of the `x64` directory. It also means that the `PATH` would not be set at all (for .NET).
+- If x64 builds of Visual Studio for Windows or Visual Studio for Mac are installed on Arm64, then they will need to install the Arm64 .NET SDK and/or adapt those products to .NET x64 being installed in the `x64` directory.
+
+We could install .NET x64 to another location, but the implications would be the same.
+
+## Model for targeting architecture
+
+Goal: Enable developers to produce architecture-specific assets correctly and with confidence.
+
+The .NET application model is oriented around rid-specific apps. In particular, the apphost is rid-specific, and the apphost ia a core part of the experience. In some scenarios, you don't have to pay much attention to the rid of the apphost. For example, if you exclusively develop on *and* target Windows x64, then rid-targeting isn't really important, even though it is present. If your development and target environment differ, then you need to directly participate in rid-targeting.
+
+In the typical scenario, you can develop on your machine, for example Windows x64, and then not need to consider rid-targeting until you are ready to test on or deploy to Linux x64, for example. x64 emulation places new requirements on developers to participate in RID-targeting during inner loop development. That's new.
+
+The core issue is that some .NET versions are available for one rid and not the other. That means that you need to explicitly target the matching rid in some way. There are three choices for that, two of which are already supported today and one not.
+
+### Rely on the implicit rid of the SDK
+
+In theory, the easiest approach is to simply use the matching SDK for the rid you want to target. That's the model used in the "developing and targeting on Windows x64" example discussed earlier.
```

Comment:
What is the "developing and targeting on Windows x64" example that this talks about?

Reply:
It just means that you develop on and Windows x64 and deploy to Windows x64 for prod, rid targeting isn't something you need to think about. Same thing with Linux x64 (for development and prod). You need to think more about rid targeting when the development environment and prod environment don't match in operating system and/or architecture dimensions. Make sense?

Reply:
Yes, makes sense. I was commenting more on the fact that this example is not actually discussed earlier in this doc, and so people reading this doc for the first time will be lost.

Reply:
Fixed.